clippy_lints/src/transmute/mod.rs

```diff
@@ -253,6 +254,30 @@ declare_clippy_lint! {
     "transmutes from an integer to a float"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for transmutes from integers to non-zero wrapper types, and suggests their
+    /// `new_unchecked` method instead.
+    ///
+    /// ### Why is this bad?
+    /// Transmutes are dangerous and error-prone, whereas `new_unchecked` is more appropriate here.
+    ///
+    /// ### Example
+    /// ```rust
+    /// # use core::num::NonZeroU32;
+    /// let _non_zero: NonZeroU32 = unsafe { std::mem::transmute(123) };
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// # use core::num::NonZeroU32;
+    /// let _non_zero = unsafe { NonZeroU32::new_unchecked(123) };
+    /// ```
+    #[clippy::version = "1.69.0"]
+    pub TRANSMUTE_INT_TO_NON_ZERO_WRAPPER,
```

Comment:
I'd shorten that to `TRANSMUTE_INT_TO_NON_ZERO` .

Reply:
Done.